velox/functions/prestosql/aggregates/PrestoHasher.cpp

```diff
@@ -135,6 +135,11 @@ FOLLY_ALWAYS_INLINE void PrestoHasher::hash<TypeKind::BIGINT>(
       // returns the corresponding value directly.
       return vector_->valueAt<int64_t>(row);
     });
+  } else if (isTimestampWithTimeZoneType(vector_->base()->type())) {
+    // Hash only timestamp value.
```

Comment:
I understand that this is how it was before, but why do we ignore TZ part?

Reply:
Hi @spershin, I believe it was because Presto Java also ignores TZ part and we followed that behavior (https://github.com/prestodb/presto/blob/36575b0ca7d61a0ecf0be8eac7efc0b2b0d3ec85/presto-common/src/main/java/com/facebook/presto/common/type/TimestampWithTimeZoneType.java#L51-L54). I found a relevant discussion here: https://github.com/prestodb/presto/issues/17274.